Core/GDCore/Project/Project.cpp

```diff
@@ -736,6 +751,18 @@ void Project::SerializeTo(SerializerElement& element) const {
     authorIdsElement.AddChild("").SetStringValue(authorId);
   }
 
+  auto& playableDevicesElement = propElement.AddChild("playableDevices");
+  playableDevicesElement.ConsiderAsArray();
+  if (isPlayableWithKeyboard) {
+    playableDevicesElement.AddChild("").SetStringValue("keyboard");
+  }
+  if (isPlayableWithGamepad) {
+    playableDevicesElement.AddChild("").SetStringValue("gamepad");
+  }
+  if (isPlayableWithMobile) {
+    playableDevicesElement.AddChild("").SetStringValue("mobile");
+  }
+
   // Compatibility with GD <= 5.0.0-beta101

```

Comment:
You also need to add the 3 booleans to Project::Init (to make sure they are copied when a copy in C++ is done - ask me if the reason for this method to exist is unclear to you and I or Alexandre/Clement can explain)

Reply:
Should I also set a default value in the constructor?
Not every attributes are initialized and I'm not sure how to choose.